### DISCUSSION TITLE
Zero to hero CoppeliaSim

### DISCUSSION THREAD

**Discussion by ffasilva** [dqrobotics/learning-dqrobotics-with-coppeliasim#1] [Ideas]
@bvadorno e @mmmarinho ,

O @juanjqo e eu discutimos sobre o tutorial e tivemos a idea de criar os seguintes exercícios:

1. CoppeliaSim/V-REP basics
   - How to select engines;
   - How to change engine parameters (gravity, simulation step, etc.)
2. Static world (part 1)
   - How to interact with objects (get/set pose, etc.)
3. Static world (part 2)
   - How to interact with robots (get/set joint positions, etc.)
4. Dynamic world 
   - How to interact with robots (get/set joint positions, etc.)

A ideia é criarmos duas cenas:
- _static_world.ttt_: objects + robot manipulator
- _dynamic_world.ttt_: robot manipulator + mobile base

O que vocês acham?

P.S.: No futuro podemos incluir atuação/leitura de torques no tutorial 4, mas essa interface ainda não existe na versão atual da _DQ_VrepInterface_.

**Comment by mmmarinho** [comment]
@ffasilva 

Oba, legal. Sobre essa parte abaixo, uma dúvida rápida.

> P.S.: No futuro podemos incluir atuação/leitura de torques no tutorial 4, mas essa interface ainda não existe na versão atual da _DQ_VrepInterface_.

Esses metodos abaixo, por exemplo, não estão funcionando para esse propósito? Se eu me lembro bem eles ainda não estão disponíveis no MATLAB, mas o `{get,set}_joint_velocities()` aparentemente também não...

https://github.com/dqrobotics/cpp-interface-vrep/blob/93ccb967a2aa6ba6f14cd6a8cc7c77df009c6ffd/include/dqrobotics/interfaces/vrep/DQ_VrepInterface.h#L179

https://github.com/dqrobotics/cpp-interface-vrep/blob/93ccb967a2aa6ba6f14cd6a8cc7c77df009c6ffd/include/dqrobotics/interfaces/vrep/DQ_VrepInterface.h#L182

**Reply by ffasilva** [reply]
A ideia do tutorial é ser no MATLAB, não? Por isso eu não estava considerando as funções do CPP. 

Sobre o `{get,set}_joint_velocities()`, não havia reparado que eles também não existem na interface em MATLAB. Vou editar a discussão para removê-los também.

**Comment by juanjqo** [comment]
@mmmarinho não, esses métodos não estão disponíveis ainda em dqrobotics/matlab. Estou trabalhando nisso #UnderDevelopment :neckbeard:
Por enquanto podemos colocar no tutorial o que estiver disponível.

**Comment by bvadorno** [comment]
Hi @dqrobotics/developers,

I'm writing in English because this discussion will potentially become public afterward. The lessons are sensible for a first start. I'd also suggest something containing synchronous and asynchronous modes with a brief discussion of their differences.

@juanjqo, we might treat the inclusions of those functions as a high priority. Maybe @ffasilva might help you with some of your DQ Robotics stuff to alleviate your load. What do you both think?

All the best,
Bruno

**Reply by juanjqo** [reply]
@bvadorno Yeah, I will prepare a PR including those functionalities this week. Concerning the additional @ffasilva's help, Sure! All help is welcome!
 
 Best regards,

Juan

**Reply by ffasilva** [reply]
I agree. We are missing too many essential features in the current _DQ_VrepInterface_ on MATLAB. It will be better if we add them before releasing the tutorial. @juanjqo, let me know how I can help you!
